joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/passes/DownloadDependenciesPass.scala

```diff
@@ -25,21 +28,25 @@ class DownloadDependenciesPass(parentGoMod: GoModHelper, goGlobal: GoGlobal, con
   private def setupDummyProjectAndDownload(prjDir: String): Unit = {
     parentGoMod
       .getModMetaData()
-      .map(mod => {
+      .foreach(mod => {
         ExternalCommand.run("go mod init joern.io/temp", prjDir) match
           case Success(_) =>
-            mod.dependencies
-              .filter(dep => config.includeIndirectDependencies || !dep.indirect)
-              .foreach(dependency => {
-                val dependencyStr = s"${dependency.module}@${dependency.version}"
-                val cmd           = s"go get $dependencyStr"
-                ExternalCommand.run(cmd, prjDir) match
-                  case Success(_) =>
-                    print(". ")
-                    processDependency(dependencyStr)
-                  case Failure(f) =>
-                    logger.error(s"\t- command '${cmd}' failed", f)
+            val futures = mod.dependencies
+              .filter(dep => !dep.beingUsed)
+              .map(dependency => {
+                Future {
+                  val dependencyStr = s"${dependency.module}@${dependency.version}"
```

Comment:
I suppose `go get` will handle this, but one issue that comes to mind is rate limiting of the dependency APIs

Reply:
Would also be good to have a version of this where the dependencies are downloaded directly from the APIs like C# and Ruby so that we don't require `go` as a dependency

Reply:
> I suppose go get will handle this, but one issue that comes to mind is rate limiting of the dependency APIs


`go get` doesn't parallelise it, I did a test to check whether it improves the time it takes to process. I do see the improvements.

Reply:
> Would also be good to have a version of this where the dependencies are downloaded directly from the APIs like C# and Ruby so that we don't require go as a dependency

I think that will take time, I will handle that through a separate ticket.